sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobInputOutputStreamTest.groovy

```diff
@@ -121,14 +124,49 @@ class BlockBlobInputOutputStreamTest extends APISpec {
         3 * Constants.MB || 2         | [3 * Constants.MB, 3 * Constants.MB] // Smaller than default
     }
 
+    // Only run this test in live mode as BlobOutputStream dynamically assigns blocks
+    @LiveOnly
+    def "BlobRange"() {
+        setup:
+        int length = 6 * Constants.MB
+        byte[] randomBytes = getRandomByteArray(length)
+
+        BlobOutputStream outStream = bc.getBlobOutputStream(true)
+        outStream.write(randomBytes, 0, 6 * Constants.MB)
+        outStream.close()
+
+        def resultBytes = new byte[count == null ? length - start : count]
+        when:
+        def inputStream = bc.openInputStream(new BlobInputStreamOptions().setRange(new BlobRange(start, count))
+            .setBlockSize(4 * Constants.MB))
+        inputStream.read(resultBytes) // read the whole range
+
+        then:
+        inputStream.read() == -1
+        ByteBuffer.wrap(randomBytes, start, count == null ? length - start : count)  == ByteBuffer.wrap(resultBytes)
+
+        where:
+        start            | count
+        0                | null // full blob
+        0                | 100 // Small range
+        0                | 4 * Constants.MB // block size
+        0                | 5 * Constants.MB // Requires multiple chunks
+        5                | 100 // small offset
+        5                | null // full blob after an offset
+        1 * Constants.MB | 2 * Constants.MB // larger offset inside first chunk
+        1 * Constants.KB | 4 * Constants.MB // offset with range spanning chunks
+        5 * Constants.MB | (1 * Constants.KB) // Range entirely in second chunk
+        5 * Constants.MB | null // rest of blob after first chunk
```

Comment:
can we also throw in some cases where boundaries are not a multiple of 1024 ?

Reply:
Added